lib/goth/token.ex

```diff
@@ -52,9 +57,10 @@ defmodule Goth.Token do
         {Goth.HTTPClient.Hackney, Goth.HTTPClient.Hackney.init([])}
       end)
 
-    jwt = jwt(config.scope, config.credentials)
+    jwt_or_metadata = jwt(config.scope, config.credentials)
+    token_scope = maybe_override_token_scope(jwt_or_metadata, config.scope)
 
-    case request(config.http_client, config.url, jwt) do
+    case request(config.http_client, config.url, jwt_or_metadata) do
```

Comment:
instead of two sets of clauses (jwt/3+instance & request/3+metadata and jwt/3 & request/3) how about we have multiple clauses of request/1? Something like this:

```elixir
defp request(%{credentials: {:instance, account}} = config) do
  headers = [{"metadata-flavor", "Google"}]
  ...
  Goth.HTTPClient.request(http_client, :get, url, headers, "", [])
end

defp request(config) do
  headers = [{"content-type", "application/x-www-form-urlencoded"}]
  grant_type = "urn:ietf:params:oauth:grant-type:jwt-bearer"
  jwt = jwt(config.scope, config.credentials)
  body = "grant_type=#{grant_type}&assertion=#{jwt}"
  HTTPoison.request(:post, url, body, headers, opts)
end
```

and then if we do overrides, we can even inline them in the appropriate clause.